### PR TITLE
Add new tool rasusa for read subsampling

### PR DESCRIPTION
**Pull request by pavanvidem**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/rasusa/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+categories:
+- Sequence Analysis
+description:  Randomly subsample sequencing reads to a specified coverage
+homepage_url: https://github.com/mbhall88/rasusa
+long_description: |
+  Randomly subsample sequencing reads to a specified coverage from FASTQ or FASTQ files.
```
Comment (bernt-matthias) [inline tools/rasusa/.shed.yml:6]: > from FASTQ or FASTQ files

2x fastq?

File: tools/rasusa/rasusa.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+<tool id="rasusa" name="rasusa" version="0.8.0+galaxy0" python_template_version="3.5" profile="21.05">
```
Comment (bernt-matthias) [inline tools/rasusa/rasusa.xml:1]: Use a version token here and in the requirements.

File: tools/rasusa/rasusa.xml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    ]]></command>
+    <inputs>
+        <conditional name="fastq_input">
+            <param name="fastq_input_selector" type="select" label="Single or Paired-end reads" help="Select between paired and single end data">
+                <option value="paired">Paired</option>
+                <option value="single">Single</option>
+                <option value="paired_collection">Paired Collection</option>
+            </param>
+            <when value="paired">
+                <param name="fastq_input1" type="data" format="fastq,fasta" label="Select first set of reads" help="Specify dataset with forward reads"/>
```
Comment (bernt-matthias) [inline tools/rasusa/rasusa.xml:36]: Is zipped input supported?

File: tools/rasusa/rasusa.xml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            <when value="by_num">
+                <param argument="num" type="integer" value="" min="0"/>
+            </when>
+            <when value="by_frac">
+                <param argument="frac" type="float" value="" min="0" max="1"/>
+            </when>
+        </conditional>
+        <param type="text" argument="genome-size" label="Genome size to calculate coverage with respect to"/>
+        <param type="text" argument="coverage" label="The desired coverage to sub-sample the reads to"/>
+        <param type="text" argument="bases" label="Explicitly set the number of bases required"/>
```
Comment (bernt-matthias) [inline tools/rasusa/rasusa.xml:60]: For argument you should use the command line option.

File: tools/rasusa/rasusa.xml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        <param type="integer" argument="seed" optional="true" label="Random seed to use"/>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data name="output" format="fastq,fasta" label=""/>
+    </outputs>
```
Comment (bernt-matthias) [inline tools/rasusa/rasusa.xml:66]: Please add some tests.

File: tools/rasusa/rasusa.xml [hunk 6]
```diff
@@ -0,0 +1,261 @@
+<tool id="rasusa" name="rasusa" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" python_template_version="3.5" profile="21.05">
```
Comment (bgruening) [inline tools/rasusa/rasusa.xml:1]:
```suggestion
<tool id="rasusa" name="rasusa" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@"  profile="22.05">
```

File: tools/rasusa/rasusa.xml [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+            </conditional>
+            <param name="seed" value="1"/>
+            <output_collection name="collection_output" type="paired">
+                <element name="forward" file="paired1_by_frac_reads.fasta" ftype="fasta"/>
+                <element name="reverse" file="paired2_by_frac_reads.fasta" ftype="fasta"/>
+            </output_collection>
+        </test>
+    </tests>
+    <help><![CDATA[
+        Randomly subsample reads to a specified coverage
```
Comment (bgruening) [inline tools/rasusa/rasusa.xml:256]: Can you describe this a bit more?

**Review by bgruening — APPROVED** [review]

**Comment by bernt-matthias** [comment]
Restarted and deployed..

**Comment by pavanvidem** [comment]
thanks @bernt-matthias and @bgruening

**Pull request merged**